pytket/extensions/qiskit/qiskit_convert.py

```diff
@@ -716,7 +743,7 @@ def _append_if_else_circuit(
     else:
         raise TypeError(
```

Comment:
According to the [qiskit docs](https://quantum.cloud.ibm.com/docs/en/api/qiskit/qiskit.circuit.IfElseOp) this can also be an `expr.Expr`, so perhaps we should assert that and raise a `NotImplementedError` for this case.

Reply:
Good point. I've added it here to be sure [6e5eeac](https://github.com/CQCL/pytket-qiskit/pull/524/commits/6e5eeac99fe5999e39a634b924a7939857055c52). 

Its not super clear from the source but I was under the impression that `Var`, `Binary` and `Unary` inherit from `Expr`. Can't hurt to check for it explicitly though.